aws/aws-toolkit-vscode
# Initial flow for entering credentials profile

## Types of changes


## Description

* When a user tries to connect to AWS, and no credentials profiles can be found...
  * if they have no credentials/config file, they are prompted to enter credentials
  * if they have credentials and/or config files, these are opened in the editor for the user to modify
* New Command Palette action allows user to Create a credential profile, which follows the same flow as above
* When user enters credentials through a prompt, they are validated with a call to STS getCallerIdentity
  * Credentials that do not appear to be valid will prompt the user if they would like to enter credentials again

## Motivation and Context

Reduce friction for new users to get up and running with AWS resources 

## Related Issue(s)

#56 

## Testing

* "New user"
  * renamed my current `credentials` and `config` files
  * Select `Connect to AWS` from the Command Palette
  * Expect to be asked if I want to enter credentials
  * Entering valid credentials should write out a template-generated `credentials` file
  * Entering invalid credentials should ask the user if they want to try again
* "New user" part 2
  * renamed my current `credentials` and `config` files
  * Select `Create Credentials Profile` from the Command Palette
  * Entering valid credentials should write out a template-generated `credentials` file
  * Entering invalid credentials should ask the user if they want to try again
* "Credentials file(s) without Profiles"
  * Set up credentials and config files containing no profiles (two situations: one time with credentials file only, and another time with both files)
  * Select `Connect to AWS` from the Command Palette
  * Expect to be asked if I want to enter credentials
  * Expect files to be open in editor
* "Credentials file with Profiles"
  * Have normal credentials files
  * Select `Create Credentials Profile` from the Command Palette
  * Expect files to be open in editor





## Screenshots (if appropriate)

Try to connect without any credentials files. Also demonstrates what happens entering invalid credentials:

![enter-new-credentials-and-validate](https://user-images.githubusercontent.com/39839589/46702824-753bdf80-cbd9-11e8-8c47-29967728fe41.gif)

Try to connect when you have a credentials file but no (valid) profiles defined in it:

![credentials-file-no-profile](https://user-images.githubusercontent.com/39839589/46702841-897fdc80-cbd9-11e8-8906-680d481426b1.gif)

Try to create a credentials profile when you already have credentials and config files. Both open up side by side:

![open-both-credential-files](https://user-images.githubusercontent.com/39839589/46702863-a0263380-cbd9-11e8-92d0-2c31ea59272d.gif)




## Checklist





## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
